frontend/src/lib/utils.ts

```diff
@@ -46,11 +45,73 @@ export function debounce(delay: number, fn: any): any {
 }
 
 /**
- * Returns true if the URL parameters indicated that we're embedded in an
- * iframe.
+ * Param values, which can be set in ?embed={value}, all should be lowercase
  */
-export function isEmbeddedInIFrame(): boolean {
-  return url.parse(window.location.href, true).query.embed === "true"
+export const embedParamValues = new Set<string>([
+  "noColoredBar".toLowerCase(),
+  "noToolBar".toLowerCase(),
+  "noPadding".toLowerCase(),
+  "noScroll".toLowerCase(),
+  "noFooter".toLowerCase(),
+  "true".toLowerCase(),
+])
+
+/**
+ * Returns list of defined embed url params
+ */
+export function getEmbedUrlParams(): Set<string> {
+  const embedUrlParams = new Set<string>()
+  const urlParams = new URLSearchParams(window.location.search)
+  urlParams.forEach((paramValue, paramKey) => {
+    paramKey = paramKey.toString().toLowerCase()
+    paramValue = paramValue.toString().toLowerCase()
+    if (paramKey === "embed" && embedParamValues.has(paramValue)) {
+      embedUrlParams.add(paramValue)
+    }
+  })
+  return embedUrlParams
+}
+
+/**
+ * Returns true if the URL parameters contain ?embed=true (case insensitive).
+ */
+export function hasEmbedUrlParamSetToTrue(): boolean {
+  return getEmbedUrlParams().has("true")
+}
+
+/**
+ * Returns true if the URL parameters contain ?embed=noColoredBar (case insensitive).
+ */
+export function hasEmbedUrlParamSetToNoColoredBar(): boolean {
+  return getEmbedUrlParams().has("noColoredBar".toLowerCase())
```

Comment:
We are repeating string names here and inside the `embedParamValues`. Maybe it would be worth to extract names to constants?

Reply:
This and renaming to simpler names is a great advice, thank you 🙇 will do!